src/parser.ts

```diff
@@ -2913,32 +2913,9 @@ export class Parser extends DiagnosticEmitter {
     var startPos = tn.tokenPos;
     var precedence = determinePrecedenceStart(token);
     if (precedence != Precedence.NONE) {
-      let operand: Expression | null;
-
       // TODO: SpreadExpression, YieldExpression (currently become unsupported UnaryPrefixExpressions)
-
-      // NewExpression
-      if (token == Token.NEW) {
-        operand = this.parseExpression(tn, Precedence.CALL);
-        if (!operand) return null;
-        if (operand.kind == NodeKind.CALL) {
-          return Node.createNewExpression(
-            (<CallExpression>operand).expression,
-            (<CallExpression>operand).typeArguments,
-            (<CallExpression>operand).arguments,
-            tn.range(startPos, tn.pos)
-          );
-        } else {
-          this.error(
-            DiagnosticCode.Operation_not_supported,
-            tn.range()
-          );
-        }
-        return null;
-      } else {
-        operand = this.parseExpression(tn, precedence);
-        if (!operand) return null;
-      }
+      const operand = this.parseExpression(tn, precedence);
```

Comment:
This will result in a warning when compiling to WASM, because `operand` isn't a compile-time constant. Certainly not really JS-y, but at this point `const` is a bit special in AS.